dynamixel_hardware/src/dynamixel_hardware.cpp

```diff
@@ -382,13 +384,19 @@ return_type DynamixelHardware::set_control_mode(const ControlMode & mode, const
       }
     }
     RCLCPP_INFO(rclcpp::get_logger(kDynamixelHardware), "Velocity control");
-    control_mode_ = ControlMode::Velocity;
+    if(control_mode_ != ControlMode::Velocity)
+    {
+      mode_changed_ = true;
```

Comment:
I think it's better to locate the following code at the first line of `DynamixelHardware::set_control_mode` instead.
```
mode_changed_ = mode != control_mode_;
```

Reply:
If an attempt is made to change to an unsupported mode (eg Effort), your suggested change would result in mode_changed_ being set to true.  My intent was to only set it true when the mode change was successful.

Reply:
Oh, I see. It's nice.

Reply:
If an attempt is made to change to an unsupported mode (eg Effort), your suggested code would result in mode_changed_ being set to true. My intent was to only set it true when the mode change was successful.  What do you think ?